src/Paramore.Brighter.MessagingGateway.AWSSQS/SnsMessageProducer.cs

```diff
@@ -173,8 +150,46 @@ public void SendWithDelay(Message message, TimeSpan? delay = null)
     /// <exception cref="NotImplementedException"></exception>
     public async Task SendWithDelayAsync(Message message, TimeSpan? delay,
         CancellationToken cancellationToken = default)
-    {
-        // SNS doesn't support publish with delay
-        await SendAsync(message, cancellationToken);
+    { 
+        delay ??= TimeSpan.Zero;
+        if (delay != TimeSpan.Zero)
+        {
+            if (Scheduler is IAmAMessageSchedulerAsync async)
+            {
+                await async.ScheduleAsync(message, delay.Value, cancellationToken);
```

Comment:
Probably best to avoid a keyword in a variable name, use asyncScheduler instead?